snapcraft/plugins/rust.py

```diff
@@ -193,21 +194,74 @@ def _get_target(self) -> str:
             )
         return rust_target.format("unknown-linux", "gnu")
 
+    def _project_uses_workspace(self) -> bool:
+        path = Path(self.builddir, "Cargo.toml")
+        if not path.is_file():
```

Comment:
can anything build without a Cargo.toml? Maybe an exception about not being a supported rust project would be better?

Reply:
I don't think it's valid without Cargo.toml according my minimal experience and testing, but I don't know that's saying much :smile: 

While I'm rather OK with the `cargo install` step failing after this as it should be rather apparent to the user when cargo complains (`error: <path> does not contain a Cargo.toml file. --path must point to a directory containing a Cargo.toml file.`), I do think it's reasonable to at least warn on conditions in which we believe the plugin will fail, particularly if it ends up being more helpful for the user to diagnose.

Is there an existing exception you would prefer? Or perhaps just `warn()`? Or add a new one like `SnapcraftPluginAssertionFailure()` so we can use it across all plugins consistently?

Reply:
Failing here is fine, if it ever works it will work by chance, right?

We have multiple in-plugin exceptions already, but I am not opposed to a generic one